src/pages/index.js

```diff
@@ -1,87 +1,237 @@
 
-// const newCardPopup = new PopupWithForm(constants.selectors.addModal, () =>
+// const newCardPopup = new PopupWithForm(addCardModal, () =>
 //   console.log("clicked")
 // );
 // newCardPopup.open();
 
 // newCardPopup.close();
 
-const cardPreview = new PopupWithImage("#preview-image-modal");
-cardPreview.setEventListeners();
+/*Functions*/
 
-function handleImageClick(name, link) {
-  cardPreview.open({ name, link });
+function openPreviewModal(data) {
+  previewImagePopup.open(data);
 }
 
-function createCard(cardData) {
-  const cardElement = new Card(cardData, "#card-template", handleImageClick);
-  return cardElement.getView();
+/* Instantiations*/
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "d5d1d874-a161-48bd-b3b4-ae8e38400ed9",
+    "Content-Type": "application/json",
+  },
+});
+
+let cardSection;
+
+api
+  .getUserInfo()
+  .then((res) => {
+    userInfo.setUserInfo(res);
+    userInfo.setUserAvatar(res);
+  })
+  .catch((err) => {});
+
+api
+  .getInitialCards()
+  .then((res) => {
+    cardSection = new Section(
+      { items: res, renderer: renderCard },
+      ".cards__list"
+    );
+    cardSection.renderItems();
+  })
+  .catch((err) => {
+    console.log(err);
+  });
+
+function renderCard(cardData) {
+  const card = new Card(
+    cardData,
+    selectors.cardTemplate,
+    openPreviewModal,
+    handleCardDeleteClick,
+    handleLikeClick
+  ).getCardElement();
+  cardSection.addItem(card);
 }
 
-const userInformation = new UserInfo({
+const userInfo = new UserInfo({
   name: ".profile__title",
   description: ".profile__description",
+  userAvatarSelector: ".profile__avatar",
 });
 
-const profileEditForm = new PopupWithForm("#profile-edit-modal", (data) => {
-  console.log("Form data", data);
-  userInformation.setUserInfo({
-    title: data.title,
-    description: data.description,
+const profileEditForm = new PopupWithForm(
+  "#profile-edit-modal",
+  handleProfileEditSubmit
+);
+const addCardPopup = new PopupWithForm("#add-card-modal", handleCardFormSubmit);
+const previewImagePopup = new PopupWithImage("#modal__preview-card");
+const confirmDeletePopup = new PopupWithConfirm("#delete-card-modal");
+
+const avatarImagePopup = new PopupWithForm(
+  "#edit-avatar-modal",
+  handleAvatarFormSubmit
+);
+
+/*Validation*/
+
+const editFormValidator = new FormValidator(config, profileForm);
+const addCardValidator = new FormValidator(config, cardForm);
+const avatarValidator = new FormValidator(config, avatarUpdateForm);
+
+/*Event Handlers*/
+
+function handleProfileEditSubmit(inputItems) {
+  profileEditForm.renderLoading(true);
+  api
+    .userProfileInfo(inputItems.name, inputItems.about)
+    .then((res) => {
+      console.log(res);
+      userInfo.setUserInfo(res);
+      profileEditForm.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      profileEditForm.renderLoading(false);
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}

```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```